libtextsecure/message_receiver.js

```diff
@@ -1066,7 +1066,7 @@ MessageReceiver.prototype.extend({
         primaryDevicePubKey,
         secondaryDevicePubKey,
         dcodeIO.ByteBuffer.wrap(requestSignature).toArrayBuffer(),
-        type
+        textsecure.protobuf.PairingAuthorisationMessage.Type.REQUEST
```

Comment:
Not sure I follow here
You are verifying *all* authorisations as REQUESTS
And then if `isGrant` is true you are *also* verifying them as GRANTS?

Reply:
I agree this is not super clear.
Here's a simplified explanation:
If the message type is `REQUEST`, we should `verify(requestSignature, REQUEST)`.
And if message type is `GRANT`, we should `verify(requestSignature, REQUEST) && verify(grantSignature, GRANT)`

Reply:
Ah right, ty for clarification